conda/environment.yml

```diff
@@ -3,27 +3,164 @@
-    - jinja-markdown==1.210911
+  # dashboards
+  - voila~=0.5
+
+  # Core scientific python stack
+  - cython~=3.0.10 # optimization, C API access
+  - flox~=0.9.7 # optimization, xarray operations
+  - hypothesis==6.58.0 # needed by numpy testing tools
+  - networkx==2.8.8
+  - numba~=0.60.0 # high-performance numerics
+  - numpy==1.26.4
+  - pandas>=2.0.1,<3.0
+  - scikit-image~=0.24.0
+  - scikit-learn>=1.2.2,<2.0.0
+  - scipy>=1.10
+  - statsmodels==0.13.5
+  - sympy~=1.12
+  - xarray>=2024.05.0
+
+  # Visualization packages
+  - bokeh~=3.2.2
+  - cartopy~=0.23.0 # geospatial plotting with matplotlib
+  - geemap~=0.32
+  - geoviews~=1.10.1
+  - hvplot~=0.10.0
+  - ipyleaflet~=0.19.2
+  - ipympl~=0.9.3 # This enables matplotlib interaction with jupyter widgets
+  - leafmap~=0.36
+  - lonboard~=0.9
+  - matplotlib~=3.9.0
+  - plotly~=5.18.0
+  - seaborn~=0.13.2 # statistical plotting with matplotlib
+
+  # Geospatial data packages
+  - geopandas>=0.14.4
+  - pygmt~=0.12.0
+  - rasterio~=1.3
+  - rioxarray~=0.17.0
+
+  # File formats and file management, download, dataset caching
+  - h5py~=3.11
+  - h5netcdf~=1.3.0
+  - nco~=5.1.6
+  - pooch~=1.8.0
+  - zarr~=2.18
+  - kerchunk~=0.2
+
+  # Cloud access tools and libraries
+  - awscli~=1.27.76
+  - boto3
+  - s3fs>=2023.6.0
+  # handle login and similar details for accessing earthdata protected data
+  # https://github.com/nsidc/earthaccess
+  - earthaccess~=0.10.0
+  # Access datasets exposed via STAC
+  - pystac-client~=0.8.3
+  # Access datasets exposed via intake catalogs
+  - intake==0.6.6
+  - intake-esm>=2023.7.7
+  - intake-stac==0.4.0
+  - intake-xarray==0.6.1
+  - gcsfs>=2023.5.0
+
+  # Distributed computing
+  - dask>=2023.12.1
+  - dask-labextension~=7.0.0
+  - dask-geopandas~=0.4.1
+
+  # Other useful generic python packages
+  - pillow~=10.2.0 # Python imaging library, useful for many image-related tasks
+  - pytest~=8.3
+  - pytest-cov~=5.0
+  - pep8~=1.7
+  - flake8~=7.1
+  - tqdm~=4.66 # progress bars, with notebook support
+
+  # Packages specific to climate work
+  - esmpy~=8.4.2
+  - xmip~=0.7.1
+
+  # Packages specific to cryo work
+  - h5coro~=0.0.6
+  - icepyx~=1.1.0
+  - itslive~=0.3.2
+  - is2view~=0.0.8
+  - sliderule~=4.5
+  - snowexsql~=0.4
+
+  # Desktop tools whose versions are more recent on conda-forge than ubuntu
+  #- qgis~=3.38.0
+  # Maybe needed for qgis? https://github.com/conda-forge/qgis-feedstock/issues/263
+  #- pyopencl
+  # Resolves warning "No ICDs were found": https://github.com/CryoInTheCloud/hub-image/issues/50
+  #- ocl-icd-system
+  # needed for Linux desktop to work
+  #- websockify>=0.10
+  # Includes new subcommands (`switch` & `restore`) and vulnerability fixes
+  # over the version available via ubuntu sources
+  #- git>=2.39
+
+  # Not all packages will be available from conda-forge, we install from pip when we need to.
+  #- pip~=23.3.2
+  #- pip:
+    # Access linux desktop from inside JupyterHub
+  #  - jupyter-desktop-server==0.1.3
```

Comment:
There are some big packages here (qgis) that don't seem necessary for executing tutorial notebooks, but can bring them back if needed...